implementation/map-to-scatter-gather/README.md

```diff
@@ -0,0 +1,118 @@
+```
+
+If you are a Windows platform, you would activate the virtualenv like this:
+
+``` bash
+% .venv\Scripts\activate.bat
+```
+
+Once the virtualenv is activated, you can install the required dependencies.
+
+``` bash
+pip install -r requirements.txt
+```
+
+At this point you can now deploy the stacks.
+
+``` bash
+cdk deploy --all
+```
+
+To add additional dependencies, for example other CDK libraries, just add
+them to your `setup.py` file and rerun the `pip install -r requirements.txt`
+command.
+
+### Useful commands
+
+* `cdk ls`          list all stacks in the app
+* `cdk synth`       emits the synthesized CloudFormation template
+* `cdk deploy`      deploy this stack to your default AWS account/region
+* `cdk diff`        compare deployed stack with current state
+* `cdk docs`        open CDK documentation
+
+## Testing it out
+
+* First, lets invoke the step function workflow. Get the Step Function ARN from the Output of the cdk deploy step.(Get ARN from stack outputs: ```ScatterGatherStack.StatemachineArn```)
+
+``` bash
+aws stepfunctions start-execution --state-machine-arn arn:aws:states:[aws_region]:[account-id]:stateMachine:sfn-mapscatter-gather-workflow  --input file://scatter_gather/input.json
+```
+
+You should see an output like that
+
+``` Json
+{
+    "executionArn": "arn:aws:states:us-east-1:446084752553:execution:sfn-mapscatter-gather-workflow:bff889d4-fcc5-43d7-9bb9-94da5d8c3e95",
+    "startDate": "2023-03-29T14:53:31.469000+02:00"
+}
+```
+
+* Next, lets test the refactored version that uses SNS and SQS to implement the scatter-gather patten. We will invoke the requester lambda that triggers the execution. Get the function name from the cdk deploy output (Get function name from stack outputs:```RefactoredScatterGather.RequesterFunctionName```).
+
+``` bash
+aws lambda invoke-async --function-name RefactoredScatterGather-refactorlambdarequester8C7-Zr686IpjPlC5 --invoke-args ./scatter_gather/input.json
+```
+
+You should see:
+
+``` Json
+{
+    "Status": 202
+}
+```
+
+*Note*: We used `invoke-async`  above because Lambda Destinations only supports Asynchronous invocation.
+
+## Verify
+
+Login to your AWS console and navigate to Step Function. Look into the ```sfn-mapscatter-gather-workflow``` state machine and verify the execution. Inspect the output of the aggregator step should look similar to the following.
+
+``` Json
+{
+  "statusCode": 200,
+  "body": "[{\"uuid\": \"406ee76b-0f46-4191-9864-572dd153ccd4\", \"quote\": 616}, {\"uuid\": \"406ee76b-0f46-4191-9864-572dd153ccd4\", \"quote\": 770}, {\"uuid\": \"406ee76b-0f46-4191-9864-572dd153ccd4\", \"quote\": 330}, {\"uuid\": \"406ee76b-0f46-4191-9864-572dd153ccd4\", \"quote\": 847}]"
+}
+```
+
+Next, lets verify the refactored variant using SNS and SQS. Navigate to CloudWatch logs and select the log group of the aggregator lambda function.
+
+In the logstreams you should look for ```quotes:[{'uuid': '1ed49edb-0e9b-4e1c-9ef7-893fe7014149', 'quote': 330}]```.
```

Comment:
you could also do that from the cli with `aws logs tail` - might be simpler actually